### PR TITLE
Implemented report event api

### PR DESCRIPTION
**Pull request by flowy11**
1. Added an endpoint to report  "patient_arrived" and "practitioner_arrived" events.
2. Added EHR writeback support for changing appointment status in the EHR system.

Tested manually on AppEngine standard environment with serverless connector. Confirmed that the HL7 message can be successfully sent and received in the expected format.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -69,6 +77,16 @@ an Appengine application deployed.
 
 Deploy the application using `gcloud app deploy`.
 
+If EHR writeback is enabled, a serverless connector must be created for the app
+to access resources in a VPC network where the EHR server is connected to. Then
+in the app.yaml file, add the following section:
```
Comment (dzelemba) [inline README.md]: Which app.yaml file is this referring to?

**Review by dzelemba — COMMENTED** [review]
Looks good, just one small question.

Also, should we merge this into another feature branch? Otherwise we're submitting a partial feature to develop which makes it harder to release any bug fixes.

File: README.md (hunk 1 above)
Comment (flowy11) [reply README.md]: Added "in the app folder" statement.

**Comment by flowy11** [comment]
> Looks good, just one small question.
> 
> Also, should we merge this into another feature branch? Otherwise we're submitting a partial feature to develop which makes it harder to release any bug fixes.

The feature is partially complemented and won't affect other features. The only thing left is the validation of the EHR response and send a different message to only report "patient arrived". 
If you think it's better keep them in a feature branch, how about just using this branch (report-event-api)?  We can abandon this PR and create a new one once everything is done.

File: README.md (hunk 1 above)
Comment (dzelemba) [reply README.md]: What app folder? You mean the app.yaml file in the root directory of this repo?

Can you link to that file? Can we add a commented out section that the file?

**Comment by dzelemba** [comment]
> > Looks good, just one small question.
> > Also, should we merge this into another feature branch? Otherwise we're submitting a partial feature to develop which makes it harder to release any bug fixes.
> 
> The feature is partially complemented and won't affect other features. The only thing left is the validation of the EHR response and send a different message to only report "patient arrived".
> If you think it's better keep them in a feature branch, how about just using this branch (report-event-api)? We can abandon this PR and create a new one once everything is done.

I'd prefer to keep them in a feature branch until it's fully working and tested end to end. Otherwise we increase the chances of introducing bugs on accident. Using report-event-api SGTM, let's just make sure we're only merging commits that have gone through code review.

File: README.md (hunk 1 above)
Comment (flowy11) [reply README.md]: Done.

**Comment by flowy11** [comment]
> > > Looks good, just one small question.
> > > Also, should we merge this into another feature branch? Otherwise we're submitting a partial feature to develop which makes it harder to release any bug fixes.
> > 
> > 
> > The feature is partially complemented and won't affect other features. The only thing left is the validation of the EHR response and send a different message to only report "patient arrived".
> > If you think it's better keep them in a feature branch, how about just using this branch (report-event-api)? We can abandon this PR and create a new one once everything is done.
> 
> I'd prefer to keep them in a feature branch until it's fully working and tested end to end. Otherwise we increase the chances of introducing bugs on accident. Using report-event-api SGTM, let's just make sure we're only merging commits that have gone through code review.

Created a new branch to host the "ehr-writeback" feature and made this PR merge to that branch.
A following PR for EHR authentication check will be created and merged to the same branch.

**Pull request merged**